### ISSUE TITLE
Initialize done event

### ISSUE THREAD

**Issue by UVLabs** [fluidweb-co/fluid-checkout#123]
Do you have a JS/jQuery event that fires when your plugin has finished rendering and creating all its fields? I think ` $(document).ready(function () {})` is still too early when trying to run JS on your custom checkout page

**Comment by diegoversiani** [comment]
Hi @UVLabs,

There are no specific events triggered by Fluid Checkout, but I think you can use the WooCommerce events `init_checkout` and `updated_checkout`.

I don't know what you are trying to accomplish, but the code below is an example of using those events to change attributes of checkout fields at initialization:

```js
var set_attr_on_init = function( e ) {
	var $fields = $( 'form.checkout' ).find( 'input, select, textarea' );
	$fields.each( function() {
		// Handle attribute changes
	} );
}

$( document.body ).on( 'init_checkout', set_attr_on_init );
$( document.body ).on( 'updated_checkout', set_attr_on_init );
```

Also see my comment on another thread regarding sections being replaced by Fluid Checkout that are not replaced by WooCommerce:
https://github.com/fluidweb-co/fluid-checkout/issues/47#issuecomment-1109699262

I'm closing this topic for now, please reply to it if you need anything else.

Best,
Diego

**Comment by UVLabs** [comment]
Hi @diegoversiani Its an issue I'm having where I'm moving a field I created from one location on your checkout form to another location using the `.insertAfter()` jQuery method. Something weird was happening where the field would just get deleted from the DOM.

I thought it might be a case where the `.insertAfter()` was being trigged too early and that the field I was trying to put it after (`shipping_address_1`) was being regenerated so therefore causing the reference to the element to not actually exist.

The way I fixed this was to set a timeout for 1.5s(to be safe) before running the code that performs the move...its not a __fluid__ transition but it seems to work...

I'll try running the moving code on `updated_checkout`

**Comment by diegoversiani** [comment]
That is a long delay 😅 

1. Does it work if you have a short delay? Let's say 50ms or less?

Try using `requestAnimationFrame()` to delay it to the next "frame" and run the move code after  `updated_checkout` event has already processed, if not all, most of the associated code.

2. Is it not possible to move this code from the server side using PHP? That would be the safest way I think.
